src/Config/SettingsTab.ts

```diff
@@ -291,6 +291,68 @@ export class SettingsTab extends PluginSettingTab {
                     await this.plugin.saveSettings();
                 });
             });
+
+        // ---------------------------------------------------------------------------
+        containerEl.createEl('h4', { text: 'Reminder Settings' });
+        // ---------------------------------------------------------------------------
+
+        new Setting(containerEl)
+            .setName('Reminder Format')
+            .setDesc(
+                SettingsTab.createFragmentWithHTML(
+                    '<p class="tasks-setting-important">Do not change if reminders already exist, otherwise you will not be notified unless manually converted to new format.</p>' +
+                        '<p>The format that Tasks uses to read and write reminders time i.e 12hr 6:00 pm or 24hr 18:00.</p>',
+                ),
+            )
+            .addDropdown((dropdown) => {
+                let k: keyof typeof TIME_FORMATS;
+                for (k in TIME_FORMATS) {
+                    dropdown.addOption(TIME_FORMATS[k], k);
+                }
+                const settings = getSettings().reminderSettings;
+
+                dropdown.setValue(settings.dateTimeFormat).onChange(async (value) => {
+                    settings.dateTimeFormat = value;
+                    updateSettings({ reminderSettings: settings });
+                    await this.plugin.saveSettings();
+                });
+            });
+
+        new Setting(containerEl)
+            .setName('Daily Reminder Time')
+            .setDesc(
+                SettingsTab.createFragmentWithHTML(
+                    '<p>When daily reminders should be triggered. Should be in same format as above i.e 12 or 24hr.</p>',
+                ),
+            )
+            .addText((text) => {
+                const settings = getSettings().reminderSettings;
+                text.setPlaceholder('10')
+                    .setValue(settings.dailyReminderTime)
+                    .onChange(async (value) => {
+                        settings.dailyReminderTime = value;
+                        updateSettings({ reminderSettings: settings });
+                        await this.plugin.saveSettings();
+                    });
+            });
+
+        new Setting(containerEl)
+            .setName('Check Reminders interval')
+            .setDesc(
+                SettingsTab.createFragmentWithHTML('<p>How often Tasks should check for reminders in Seconds.</p>'),
+            )
+            .addSlider((slider) => {
+                const settings = getSettings().reminderSettings;
+                slider
+                    .setLimits(1, 30, 1)
```

Comment:
What's the thinking behind an upper limit of 30 seconds?

Reply:
Chosen a bit arbitrarily, needs less than 60 seconds. Mainly just wanted to prevent cases when users get notified too late say `hh:mm:59`. 30 seems like a good max since it would allow for two checks of where to notify or not. The original reminder plugin ran every second, which seemed a little overkill, so figured a 1-30 seconds range would be good.